be/src/exec/hdfs_scanner.cpp

```diff
@@ -473,4 +476,65 @@ bool HdfsScannerContext::can_use_dict_filter_on_slot(SlotDescriptor* slot) const
     return true;
 }
 
+void HdfsScannerContext::merge_split_tasks() {
+    if (split_tasks.size() < 2) return;
+
+    std::vector<HdfsSplitContextPtr> new_split_tasks;
+
+    auto do_merge = [&](size_t start, size_t end) {
+        auto start_ctx = split_tasks[start].get();
+        auto end_ctx = split_tasks[end].get();
+        auto new_ctx = start_ctx->clone();
+        new_ctx->split_start = start_ctx->split_start;
+        new_ctx->split_end = end_ctx->split_end;
+        new_split_tasks.emplace_back(std::move(new_ctx));
+    };
+
+    size_t head = 0;
+    for (size_t i = 1; i < split_tasks.size(); i++) {
+        bool cut = false;
+
+        auto prev_ctx = split_tasks[i - 1].get();
+        auto ctx = split_tasks[i].get();
+        auto head_ctx = split_tasks[head].get();
+
+        if ((ctx->split_start != prev_ctx->split_end) ||
```

Comment:
I'm afraid we don't have constraints to ensure the splits are consecutive. So if a new file format construct splits in a inconsecutive manner, this function will not work as expected. Do you think we should add some comments or even more strict check?

Reply:
Split ranges are consectuive, because what we do is to  split a range into pieces, so each pieces are consecutive.

What I do here is kind of defensive programming: "if split ranges are not consectuive, we can handle too".